mlflow/utils/databricks_utils.py

```diff
@@ -506,37 +543,57 @@ def get_databricks_host_creds(server_uri=None):
+        otherwise includes the hostname and authentication information necessary to
         talk to the Databricks server.
     """
-    profile, path = get_db_info_from_uri(server_uri)
-    config = ProfileConfigProvider(profile).get_config() if profile else get_config()
-    # if a path is specified, that implies a Databricks tracking URI of the form:
-    # databricks://profile-name/path-specifier
-    if (not config or not config.host) and path:
-        dbutils = _get_dbutils()
-        if dbutils:
-            # Prefix differentiates users and is provided as path information in the URI
-            key_prefix = path
-            host = dbutils.secrets.get(scope=profile, key=key_prefix + "-host")
-            token = dbutils.secrets.get(scope=profile, key=key_prefix + "-token")
-            if host and token:
-                config = DatabricksConfig.from_token(host=host, token=token, insecure=False)
-    if not config or not config.host:
-        _fail_malformed_databricks_auth(profile)
 
-    insecure = hasattr(config, "insecure") and config.insecure
+    if MLFLOW_ENABLE_DB_SDK.get():
+        from databricks.sdk import WorkspaceClient
 
-    if config.username is not None and config.password is not None:
-        return MlflowHostCreds(
-            config.host,
-            username=config.username,
-            password=config.password,
-            ignore_tls_verification=insecure,
-        )
-    elif config.token:
-        return MlflowHostCreds(config.host, token=config.token, ignore_tls_verification=insecure)
-    _fail_malformed_databricks_auth(profile)
+        profile, _ = get_db_info_from_uri(server_uri)
+        try:
+            # Using databricks-sdk to create Databricks WorkspaceClient instance,
+            # If authentication is failed, exception is raised.
+            # databricks-sdk supports many kinds of authentication ways,
+            # it will try to read authentication information by the following ways:
+            # 1. Read dynamic generated token via databricks `dbutils`.
+            # 2. parse relevant environment variables (such as DATABRICKS_HOST + DATABRICKS_TOKEN
+            #    or DATABRICKS_HOST + DATABRICKS_CLIENT_ID + DATABRICKS_CLIENT_SECRET)
+            #    to get authentication information
+            # 3. parse ~/.databrickscfg file (generated by databricks-CLI command-line tool)
+            #    to get authentication information.
+            # databricks-sdk is designed to hide authentication details and
+            # support various authentication ways, so that it does not provide API
+            # to get credential values. Instead, we can use ``WorkspaceClient``
+            # API to invoke databricks shard restful APIs.
+            WorkspaceClient(profile=profile)
+            auth_by_databricks_sdk = True
+            databricks_auth_profile = profile
+        except Exception as e:
+            _logger.info(f"Creating databricks SDK workspace client failed, error: {e!r}")
```

Comment:
This might annoy users. Imagine you're not a databricks customer and see this message every time mlflow makes an HTTP request.

Reply:
```suggestion
            _logger.info(f"Failed to create databricks SDK workspace client, error: {e!r}")
```

Reply:
question:

The code is inside `get_databricks_host_creds` function, invoking `get_databricks_host_creds` means it want to invoke databricks endpoint if I understand it correctly

Reply:
Yes, we call it when we need credentials for databricks.